packages/core-design-system/scripts/build.js

```diff
@@ -0,0 +1,304 @@
+import fs from 'node:fs/promises'
+
+import { permutateThemes, register } from '@tokens-studio/sd-transforms'
+import StyleDictionary from 'style-dictionary'
+import { getReferences, usesReferences } from 'style-dictionary/utils'
+
+import { harnessLog } from './complete-log.js'
+import {
+  COMMON_CONFIG,
+  DESIGN_SYSTEM_PREFIX,
+  DESIGN_SYSTEM_ROOT,
+  DESIGN_SYSTEM_ROOT_ESM,
+  getExportFileHeader,
+  OSS_STYLES_SOURCE_NAME,
+  STYLE_BUILD_FORMATS,
+  THEME_MODE_FILENAME_PREFIX
+} from './constants.js'
+import { generateCoreFiles, generateThemeFiles } from './sd-file-generators.js'
+
+register(StyleDictionary)
+
+async function run() {
+  let $themes
+  try {
+    $themes = JSON.parse(await fs.readFile('design-tokens/$themes.json'))
+  } catch (error) {
+    console.error('Error parsing $themes.json:', error)
+    throw error
+  }
```

Comment:
Ideally this project should use typescript instead of base Node. `ts-node` is great for that. You can then just import the JSON file directly instead of manually reading and parsing it.

As-is, there's no real benefit in a `try`/`catch` since the error is unrecoverable and thrown anyway. In this node script, we want it to die here instead of trying to be graceful.